File: py/easy/ValidParentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        for ch in s:
            if ch == "[" or ch == "(" or ch == "{":
                stack.append(ch)
            else:
                if len(stack) == 0:
                    return False
                else:
                    top = stack.pop()
                if ch == "]" and top == "[":
                    continue
                elif ch == ")" and top == "(":
                    continue
                elif ch == "}" and top == "{":
                    continue
                else:
                    return False

        return True if len(stack) == 0 else False


s = Solution()
print(s.isValid("({[]}})"))

File: ts/easy/ValidParentheses.ts
class Solution {
	isValid(s: string): boolean {
		const stack = [];
		for (const el of s) {
			switch (el) {
				case "{":
				case "[":
				case "(":
					stack.push(el);
					break;
				case "}":
					if (stack.length === 0 || stack.pop() !== '{') {
						return false;
					}
					break;
				case ")":
					if (stack.length === 0 || stack.pop() !== '(') {
						return false;
					}
					break;
				case "]":
					if (stack.length === 0 || stack.pop() !== '[') {
						return false;
					}
					break;
			}

		}

		return stack.length === 0;
	}
}
